app/gui/view/debug_scene/cloud-dashboard/src/lib.rs

```diff
@@ -37,131 +37,483 @@
+    ///
+    /// [`component`]: ::crate::projects_grid::View
+    /// [`ensogl_grid_view`]: ::ensogl_grid_view
+    /// [`Project`]: ::enso_cloud_view::project::Project
+    pub type Grid = ensogl_grid_view::simple::SimpleScrollableSelectableGridViewWithHeaders;
 
+
+
+    // ===========
     // === FRP ===
+    // ===========
 
     ensogl_core::define_endpoints_2! {
-        Input{
-            /// The component browser is displayed on screen.
-            set_size(Vector2),
+        Input {
+            set_projects(Vec<view::project::Project>),
+            model_for_entry((Row, Col), EntryModel),
         }
-        Output{
-            size(Vector2),
+        Output {
+            model_for_entry_needed((Row, Col)),
         }
     }
 
+
+
+    // =============
     // === Model ===
+    // =============
 
     #[derive(Clone, CloneRef, Debug)]
     pub struct Model {
-        root:        display::object::Instance,
-        base_shape:  rounded_box::View,
-        application: Application,
-        text:        text::Text,
+        pub root:        display::object::Instance,
+        pub grid:        Grid,
+        pub application: Application,
+        pub projects:    Rc<RefCell<Vec<view::project::Project>>>,
     }
 
+
+    // === Main `impl` ===
+
     impl Model {
         fn new(app: &Application) -> Self {
             let application = app.clone_ref();
             let root = display::object::Instance::new();
-            let text = text::Text::new(app);
-            app.display.default_scene.layers.panel.add_exclusive(&text);
+            let grid = Grid::new(app);
+            let projects = Rc::new(RefCell::new(Vec::new()));
+            root.add_child(&grid);
+
+            Self { application, root, grid, projects }
+        }
+
+        fn model_for_entry(&self, (row, col): &(Row, Col)) -> Option<(Row, Col, EntryModel)> {
+            let (row, col) = (*row, *col);
+            let projects = self.projects.borrow();
+            // The first row is the header row, which is displayed with a different entry model, so
+            // we should ignore it here.
+            if row == 0 {
+                return None;
+            }
+            // The rows of the grid are zero-indexed, but the first row is the header row, so we
+            // need to subtract 1 to get the index of the project we want to display.
+            let idx = row - 1;
+            if idx >= projects.len() {
+                warn!("Attempted to display entry at index {idx}, but we only have data up to index {}.", projects.len());
+                return None;
+            }
+            let column = match Columns::from_discriminant(col) {
+                Some(column) => column,
+                None => {
+                    warn!("Attempted to display entry at column {col}, but we the table only has {} columns.", Columns::LEN);
+                    return None;
```

Comment:
you can use `unwrap_or_else` here isntead

Reply:
Please be sure that linter is working here

Reply:
> you can use `unwrap_or_else` here isntead

`unwrap_or_else` doesn't apply cause we're returning `None`